src/lexer/token.ts

```diff
@@ -35,13 +35,21 @@ export interface Token {
   raw: string; // The raw original text that was matched
   text: string; // Cleaned up text e.g. keyword converted to uppercase and extra spaces removed
   key?: string;
+  start: number; // 0-based index of the token in the whole query string
+  end: number; // 0-based index of where the token ends in the query string
 }
 
 /**
  * For use as a "missing token"
  * e.g. in lookAhead and lookBehind to avoid dealing with null values
  */
-export const EOF_TOKEN = { type: TokenType.EOF, raw: '«EOF»', text: '«EOF»' };
+export const EOF_TOKEN: Token = {
+  type: TokenType.EOF,
+  raw: '«EOF»',
+  text: '«EOF»',
+  start: Number.POSITIVE_INFINITY,
+  end: Number.POSITIVE_INFINITY,
```

Comment:
FYI, you could have just used:

```suggestion
  start: Infinity,
  end: Infinity,
```